Rubberduck.VBEEditor/ComManagement/TypeLibs/TypeLibs.cs

```diff
@@ -0,0 +1,1669 @@
+        {
+            TypeInfoWrapper _parent;
+            public ImplementedInterfacesCollection(TypeInfoWrapper parent) => _parent = parent;
+            override public int Count { get => _parent.Attributes.cImplTypes; }
+            override public TypeInfoWrapper GetItemByIndex(int index) => _parent.GetSafeImplementedTypeInfo(index);
+
+            /// <summary>
+            /// Determines whether the type implements one of the specified interfaces
+            /// </summary>
+            /// <param name="interfaceProgIds">Array of interface identifiers in the format "LibName.InterfaceName"</param>
+            /// <param name="matchedIndex">on return, contains the index into interfaceProgIds that matched, or -1 </param>
+            /// <returns>true if the type does implement one of the specified interfaces</returns>
+            public bool DoesImplement(string[] interfaceProgIds, out int matchedIndex)
+            {
+                matchedIndex = 0;
+                foreach (var interfaceProgId in interfaceProgIds)
+                {
+                    if (DoesImplement(interfaceProgId))
+                    {
+                        return true;
+                    }
+                    matchedIndex++;
+                }
+                matchedIndex = -1;
+                return false;
+            }
+
+            /// <summary>
+            /// Determines whether the type implements the specified interface
+            /// </summary>
+            /// <param name="interfaceProgId">Interface identifier in the format "LibName.InterfaceName"</param>
+            /// <returns>true if the type does implement the specified interface</returns>
+            public bool DoesImplement(string interfaceProgId)
+            {
+                var progIdSplit = interfaceProgId.Split(new char[] { '.' }, 2);
+                if (progIdSplit.Length != 2)
+                {
+                    throw new ArgumentException($"Expected a progid in the form of 'LibraryName.InterfaceName', got {interfaceProgId}");
+                }
+                return DoesImplement(progIdSplit[0], progIdSplit[1]);
+            }
+
+            /// <summary>
+            /// Determines whether the type implements the specified interface
+            /// </summary>
+            /// <param name="containerName">The library container name</param>
+            /// <param name="interfaceName">The interface name</param>
+            /// <returns>true if the type does implement the specified interface</returns>
+            public bool DoesImplement(string containerName, string interfaceName)
+            {
+                foreach (var typeInfo in this)
+                {
+                    using (typeInfo)
+                    {
+                        if ((typeInfo.Container?.Name == containerName) && (typeInfo.Name == interfaceName)) return true;
+                        if (typeInfo.ImplementedInterfaces.DoesImplement(containerName, interfaceName)) return true;
```

Comment:
we also have `GetRefTypeOfImpType` -- should this be in here in order to search those other interfaces?

Reply:
The whole of the ImplementedInterfaces collection runs off GetRefTypeOfImplType